### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by SSEKPIUS**
Added authorization to the app using [CanCanCan](https://github.com/CanCanCommunity/cancancan).

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    #   can :update, Article, published: true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :destroy, Post do |post|
+        post.author == user
+      end
```
Comment (rotimiazeez) [inline app/models/ability.rb:34]: - Job well done here 👍🏽. Kindly implement destroy actions in the PostsController to ensure this works 😉

File: app/models/ability.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      can :destroy, Comment do |comment|
+        comment.author == user
+      end
```
Comment (rotimiazeez) [inline app/models/ability.rb:37]: - Job well done here 👍🏽. Kindly implement destroy actions in the CommentsController to ensure this works 😉

Comment (rotimiazeez) [reply app/models/ability.rb:37]: - Please ensure your `Post Controller` uses **CanCanCan** for Post authorization by adding: 👍🏽
```
load_and_authorize_resource
```

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi TEAM 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- Good teamwork 😉 
- No Linter error ✔️
- Good commit message(s) 😉
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

- Making your PR description  & summary more professional 👍🏽

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @SSEKPIUS and @Bornittah     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**